xcube/core/timecoord.py

```diff
@@ -118,17 +118,14 @@ def get_time_range_from_data(dataset: xr.Dataset, maybe_consider_metadata: bool=
     time_diff = time.diff(dim=time.dims[0]).values
     time_res = time_diff[0]
     time_regular = all([time_res - diff == np.timedelta64(0) for diff in time_diff[1:]])
+    is_cf_time = type(time.values[0]).__module__.startswith('cftime')
+    data_start = pd.to_datetime(time.values[0].isoformat()) if is_cf_time else time.values[0]
+    data_end = pd.to_datetime(time.values[-1].isoformat()) if is_cf_time else time.values[-1]
```

Comment:
Every single `time.values`  call eagerly loads all time data. I explained the implications two weeks or so ago.

```suggestion
    data_start = pd.to_datetime(time[0].values.item().isoformat()) if is_cf_time else time[0].values.item()
    data_end = pd.to_datetime(time[-1].values.item().isoformat()) if is_cf_time else time[-1].values.item()
```